### ISSUE TITLE
Nested Symbols not updating

### ISSUE THREAD

**Issue by rodi01** [DWilliames/paddy-sketch-plugin#3]
Hi,
I have a button as a symbol with with padding on it. I have another symbol with multiple buttons on it. If I try to override the text, it doesn't get updated. 

Button Symbol
![screen shot 2018-02-05 at 6 03 33 pm](https://user-images.githubusercontent.com/52461/35838237-398f8af0-0a9f-11e8-9c99-d1126f654f2c.png)

Card Symbol
![screen shot 2018-02-05 at 6 03 54 pm](https://user-images.githubusercontent.com/52461/35838246-41c32c7c-0a9f-11e8-9a1d-343510ad6d85.png)

Symbol in use
![screen shot 2018-02-05 at 6 06 51 pm](https://user-images.githubusercontent.com/52461/35838300-7e63c56a-0a9f-11e8-9e91-742c2f1af30b.png)

Thanks

**Comment by DWilliames** [comment]
At the moment Nested Symbols are unfortunately not supported — I will try and look into it more another time. I have already spent hours trying to get it to work — but it is extremely fiddly :/ 

The plugin will however, not be able to resize symbol instances within symbols. This is because Sketch currently doesn't support this. All the plugin does is resize the symbol instance to the 'optimal' size; it cannot resize the symbols within it.

**Comment by kylegilbreath** [comment]
+1 for this feature. We have components that rely pretty heavily on nested components. Would love to have this feature added. This has been our biggest pain point with keeping consistent padding on nested symbols without having to detach them from the parent symbol.

**Comment by taylorjhunt** [comment]
+1 from me as well, would be so powerful. Attached a gif of a use case. Works as a group but not as a symbol.
![number tag](https://user-images.githubusercontent.com/25557534/36178878-3865a4ee-10cf-11e8-90fc-8a86f313fc15.gif)

**Comment by mauricetimp** [comment]
+1 for this feature, would be so cool!

**Comment by anthonylgrant** [comment]
+1, would be valuable if this worked with nested symbols, linked across different libraries.

**Comment by ChrissN** [comment]
+1 for this feature, I think is a really important breaking point for this plugin, do it please and we'll give you a big thumbs up 👍  and dance out of happiness 👯‍♂️  🕺

**Comment by trvsss** [comment]
oof, this would be amazing!

**Comment by chrisdelbuck** [comment]
Just going to hop on the bandwagon here and say +1. For design system implementations, relying on having nested symbols is paramount.

**Comment by EstherW92** [comment]
+1 for this feature 😄

**Comment by MSGuzy** [comment]
Sketch 49 Enhancements included: The plugin API has been extended with support for Symbols. Hopefully, that makes this possible!

**Comment by alyseadlard** [comment]
Just another +1 for this feature. Would love to use in our Polaris UI kit! Thanks for this awesome plugin 👍

**Comment by bartvdb** [comment]
+1 .. imagine the possibilities.  Grids with symbols thats fully realign themselves when text is overwritten

**Comment by johnpeele** [comment]
Another +1

**Comment by aminta** [comment]
+1: I use a symbol for each text style, so I have only nested symbols as texts!

**Comment by johnpeele** [comment]
@aminta Would you mind explaining how you use this plugin with text symbols?  Thanks!

**Comment by Gabrielmittleman** [comment]
+1. This would be HUGE. Get this sorted and this will be the #1 essential plugin for all design systems. Great work so far!

**Comment by aminta** [comment]
@johnpeele Sure! Instead of text layers, in my "atom-design-method" I'm using - for each text style - a text symbol, so I can change them in symbol instances (i.e. I can change the text style of a button without changing the symbol but only the instance)! The joining of a such powerful plugin like "Paddy" with text styles as nested symbols would make PERFECT whole my design workflow in Sketch!

**Comment by DWilliames** [comment]
Thanks everyone!! It's great to see the support; and I'd love to get this to work. 
@MSGuzy unfortunately, that change won't actually help me out; since I was already using a different Sketch API that already supported Symbols... however; changing nested symbols dynamically within each symbol instance is a whole other ball game 🙃

Just as a quick update to this feature... looking into it more; it would basically entail a complete re-write of Paddy! Which I am totally up for doing; I just don't know how long it will take; given my limited capacity to work on this.

It would also involve doing some 'hacky' stuff to make it work. Meaning instances of these Symbols which have nested Paddy elements would not play nice with other apps such as Zeplin. Since it would involve doing some hacky stuff; it means it is quite likely that future Sketch updates may break it; so it might be a bit fiddly.

But if everyone is still up for it; given those caveats! Then I might as well get to re-writing Paddy 😉🎉

**Comment by MSGuzy** [comment]
@DWilliames I tend to agree with @fancymans, I wouldn't want something that is likely to break in a future update. Hopefully Bohemian adds overrides for text styles

**Comment by alikgs19** [comment]
+1 for this feature!!!
